compiler/rustc_next_trait_solver/src/solve/normalizes_to/mod.rs

```diff
@@ -39,11 +39,58 @@ where
             Err(NoSolution) => {
                 let Goal { param_env, predicate: NormalizesTo { alias, term } } = goal;
                 self.relate_rigid_alias_non_alias(param_env, alias, ty::Invariant, term)?;
+                self.add_rigid_constraints(param_env, alias)?;
```

Comment:
perf nit: flip order here to first check that the alias is rigid. The equate should always succeed given that `term` is an unconstrained infer var